### DISCUSSION TITLE
revokeTokensForUser Not working

### DISCUSSION THREAD

**Discussion by rodolphonetto** [adonisjs/core#1501] [Help]
Hi guys

I'm loggin a user doing this.

`const token = await auth.attempt(user.email, password, { user })`

then I catch the token and set on postman as a bearer token

then when I try to revoke it doing this

`const user = await User.findOrFail(auth.user.id)`

`await auth.authenticator('jwt').revokeTokensForUser(user)`

The token stil working, what I'm doing wrong?

**Comment by McSneaky** [comment]
JWTs can't be revoked, that's the point of JWTs :) They are totally stateless, once they are generated they can't be revoked

**Reply by rodolphonetto** [reply]
> JWTs can't be revoked, that's the point of JWTs :) They are totally stateless, once they are generated they can't be revoked

Huum... how could I add more security to them? for example a jwt that is not "eternal"? for example, my site has a logout that delete the token from the cookie but if the user save it he can use it after that.

**Reply by popovicieduard** [reply]
you can specify the expireIn for jwt's.

**Reply by rodolphonetto** [reply]
Got it... but them the user should login again right?

**Reply by popovicieduard** [reply]
Yes, after the user jwt expires, the front-end should have some sort of logic to check for the expiresIn or catch the API's exception in order to prompt the user to login again.
